packages/server-wallet/src/protocols/__test__/defunder.test.ts

```diff
@@ -37,30 +34,32 @@ async function ensureCloseObjective(
   channel: TestChannel,
   tx: Transaction,
   participantIndex = 0
-): Promise<DBCloseChannelObjective> {
-  // add a preapproved closeChannel objective
+): Promise<WalletObjective<CloseChannel>> {
+  // add the closeChannel objective and approve
   const o = await ObjectiveModel.insert(
     channel.closeChannelObjective([participantIndex, 1 - participantIndex]),
     true,
     tx
   );
-  return o as DBCloseChannelObjective;
+  await store.approveObjective(o.objectiveId, tx);
+  return o;
 }
 
 async function ensureDefundObjective(
   channel: TestChannel,
   tx: Transaction
-): Promise<DBDefundChannelObjective> {
-  // add a preapproved defundChannel objective
-  const o = await ObjectiveModel.insert(channel.defundChannelObjective(), true, tx);
-  return o as DBDefundChannelObjective;
+): Promise<WalletObjective<DefundChannel>> {
+  // add the defundChannel objective and approve
+  const o = await ObjectiveModel.insert(channel.defundChannelObjective(), false, tx);
```

Comment:
Is there a reason we can't just insert with `preApproved:true` ?

Reply:
739cdfb